index.html

```diff
@@ -2175,6 +2232,116 @@ <h3>Verify Proof Sets and Chains</h3>
+|result|.|document| to the result of running the
+<a data-cite="JSON-LD11-API#compaction-algorithm">
+JSON-LD Compaction Algorithm</a> with the modified |inputDocument| as input.
+              </li>
+            </ol>
+          </li>
+          <li>
+If no errors occurred in the previous step, set |result|.|status| to `true` and,
+if |result|.|document| is not set, set it to |inputDocument|. If errors
+occurred, set |result|.|status| to `false`, remove the |document| property from
+|result|, and add at least one error to |result|.|errors|.
+          </li>
+          <li>
+Return the value of |result|.
+          </li>
+        </ol>
+
+        <p>
+The purpose of the algorithm above is to ensure that a consuming application has
+explicitly approved of the types of input documents that it will process. Not
+checking JSON-LD Context values against known good values, and the semantics
+that they convey, can lead to security vulnerabilities. While the algorithm
+above provides one way of checking these values, and one optional way of safely
+processing unknown values, there are other approaches that provide the same
+protections.
+        </p>
+
+        <p>
+For example, if no JSON-LD processing is to occur, this check can be ignored.
+        </p>
+
+        <p>
+Another approach would be to configure an application to use a JSON-LD Context
+loader, sometimes referred to as a document loader, to use only local copies of
+approved context files which would guarantee that 1) context files will never
+change, and 2) the cryptographic hashes will never change, effectively resulting
+in the same result as the algorithm above.
+        </p>
+
+        <p>
+An alternative approach an application could use would be to use a list of well
+known context URLs and their associated approved cryptographic hashes, without
+storing every context file locally, which would allow contexts to be safely
+loaded and cached from the network without compromising the security
+expectations of the application. This approach would also be effectively
+equivalent to the algorithm provided in this section.
+        </p>
+
+        <p>
+Yet another valid approach would be for a transmitting application to
+<a data-cite="JSON-LD11-API#compaction-algorithm">compact</a> a document to
+exactly what a receiving application requests, omitting additional
+sender-specific context values that it used when securing the original document.
+As long as the cryptography suite's verification algorithm provides a successful
+verification result, such transformations are valid and would result in full
+URLs for terms that were previously compacted by the omitted context. That is, a
+term that was previously compacted to `foo` based on a sender-supplied context
+that is unknown to a receiver would instead be "compacted" to something like
+`https://vocabulary.example#foo` once the unknown context is omitted and the
+compaction algorithm is applied by the receiving application.
```

Comment:
This needs more detailed description. `foo` becoming `https://vocabulary.example#foo` looks like an expansion, not a compaction. Can we (or an update to JSON-LD docs) show a full step-by-step example, starting with a JSON-LD document that includes "unrecognized" `@context` value(s), and resulting in a JSON-LD document that has only "recognized" `@context` value(s) (including )?

Also, it is unclear here what actions are taken by the transmitting application (and whether the transmitter is a holder or issuer — the former suggested by `values that it used when securing`, which I've changed to `values that were used when securing` because I think the transmitter is *either* holder or issuer), and what actions are taken by the receiving application (only being a verifier, I think?). Significantly, I don't see any indication of how a receiving application can request specific context(s) from a transmitting application, which would be necessary for the transmitter to omit `additional sender-specific context values`...
```suggestion
Yet another valid approach would be for a transmitting application to
<a data-cite="JSON-LD11-API#compaction-algorithm">compact</a> a document to
exactly what a receiving application requests, omitting additional
sender-specific context values that were used when securing the original document.
As long as the cryptography suite's verification algorithm provides a successful
verification result, such transformations are valid and would result in full
URLs for terms that were previously compacted by the omitted context. That is, a
term that was previously compacted to `foo` based on a sender-supplied context
that is unknown to a receiver would instead be "compacted" to something like
`https://vocabulary.example#foo` once the unknown context is omitted and the
compaction algorithm is applied by the receiving application.
```